### DISCUSSION TITLE
`ImportError: cannot import name 'Mapping' from 'collections'`

### DISCUSSION THREAD

**Discussion by Mpotrue** [Lightning-AI/pytorch-lightning#14329] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I cant import lightning package:  ImportError: cannot import name 'Mapping' from 'collections'. In my Python 3.10 Mapping accessible by 'collections.abc' i.e 'from collections.abc import Mapping' not 'from collections import Mapping'. May be  someone knows how fix this issue. Import lightining package trying run next command 'from collections import Mapping'

**Comment by akihironitta** [comment, marked as answer]
Would you mind providing the full error message and your environment detail?

**Reply by Mpotrue** [reply]
> Would you mind providing the full error message and your environment detail?
Yes, of course
<img width="777" alt="image" src="https://user-images.githubusercontent.com/109548445/185791891-d86ef20b-c4aa-45ae-8c05-34c9ce3b1450.png">

**Reply by Mpotrue** [reply]
<img width="762" alt="image" src="https://user-images.githubusercontent.com/109548445/185792001-21ad1beb-edd2-4b73-93f2-79a0d1f0e16d.png">

**Reply by Mpotrue** [reply]
<img width="437" alt="image" src="https://user-images.githubusercontent.com/109548445/185792708-6a55c453-d889-4b88-bd6f-a08fa5a7e039.png">

**Reply by Mpotrue** [reply]
<img width="450" alt="image" src="https://user-images.githubusercontent.com/109548445/185792753-6acb9d25-b2ab-41ad-b91d-de33e262635d.png">

**Reply by Mpotrue** [reply]
<img width="432" alt="image" src="https://user-images.githubusercontent.com/109548445/185792790-7c9c85a5-f518-470f-bd24-de407814998a.png">

**Reply by Mpotrue** [reply]
<img width="447" alt="image" src="https://user-images.githubusercontent.com/109548445/185792808-35e0ec9b-f222-47e5-b7c6-3480f565f146.png">

**Reply by akihironitta** [reply]
The PL version installed in your env is 0.8.5 which is incompatible with Python 3.10. To use PL with Python 3.10, the version needs to be 1.7.0 or later.

**Comment by Mpotrue** [comment]
IDLE Shell has analog problem
<img width="536" alt="image" src="https://user-images.githubusercontent.com/109548445/185793058-4ea28f0e-882f-4133-8125-41cd38f9dfe8.png">

**Reply by Mpotrue** [reply]
https://docs.python.org/3.9/whatsnew/3.9.html
<img width="640" alt="Снимок экрана 2022-08-21 164347" src="https://user-images.githubusercontent.com/109548445/185793846-e81e0f4f-4be4-4833-a25a-4b61eee671de.png">

**Reply by akihironitta** [reply]
We have been aware of the change in Python 3.10 and have applied a necessary patch already. Please check the PL version installed in your environment.

**Reply by Mpotrue** [reply]
Thank you very much!

**Reply by Mpotrue** [reply]
> We have been aware of the change in Python 3.10 and have applied a necessary patch already. Please check the PL version installed in your environment.

My problem was resolved by updating PL to 1.7.2 version
